src/Tokenizer/Token.php

```diff
@@ -158,6 +158,8 @@ public function equals($other, $caseSensitive = true)
     {
         if ($other instanceof self) {
             // inlined getPrototype() on this very hot path
+            // we access the private properties of $other directly to save function call overhead.
+            // this is only possible because $other is of the same class as `self`
```

Comment:
Please fix punctuation. Hard to read when first sentence has no full stop and last sentence is not starting with upper cased letter.

Reply:
should be fixed.